cmd/cli/main.go

```diff
@@ -0,0 +1,361 @@
+							"RESTORETOS3BUCKET",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.RestoreToS3Secret,
+						Required:    false,
+						Name:        "restoreToS3Secret",
+						Usage:       "Optional ; Set restore Secret, only when using s3 restore method, expecting secret name containing key value pair with 'username' and 'password' keys, via cli or via env: ",
+						EnvVars: []string{
+							"RESTORETOS3SECRET",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.RestoreToS3SecretUsernameKey,
+						Required:    false,
+						DefaultText: "username",
+						Name:        "RestoreToS3SecretUsernameKey",
+						Usage:       "Optional ; Set RestoreToS3SecretUsernameKey, key inside secret, under which S3 username is stored, via cli or via env: ",
+						EnvVars: []string{
+							"RESTORETOS3SECRETUSERNAMEKEY",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.RestoreToS3SecretPasswordKey,
+						Required:    false,
+						DefaultText: "password",
+						Name:        "RestoreToS3SecretPasswordKey",
+						Usage:       "Optional ; Set RestoreToS3SecretPasswordKey, key inside secret, under which Restic repo password is stored, via cli or via env: ",
+						EnvVars: []string{
+							"RESTORETOS3SECRETPASSWORDKEY",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.Namespace,
+						Required:    false,
+						DefaultText: "default",
+						Name:        "namespace",
+						Usage:       "Optional ; Set namespace in which You want to execute restore, via cli or via env: ",
+						EnvVars: []string{
+							"NAMESPACE",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.Kubeconfig,
+						Required:    false,
+						DefaultText: "~/.kube/config",
+						Name:        "kubeconfig",
+						Usage:       "Optional ; Set kubeconfig to connect to cluster, via cli or via env:",
+						EnvVars: []string{
+							"KUBECONFIG",
+						},
+					},
+				},
+			},
+		},
+	}
+)
+
+func RunRestore(ctx *cli.Context) error {
+	var restoreName, snapshot string
+	var rootUser int64
+	var s3, pvc v1.RestoreMethod
+	logger := cmd.AppLogger(ctx).WithName("cli-restore")
+	// check permissions for user, defaults to 0 if not set correctly
+	rootUser, err := strconv.ParseInt(restore.Cfg.RunAsUser, 10, 64)
+	if err != nil {
+		rootUser = 0
+	}
+	// avoid name crashes + generate name if naming is not important for user
+	if restore.Cfg.RestoreName == "" {
+		restoreName = "cli-restore-" + restore.RandomStringGenerator(5)
+		log.Println("Creating restore with name: ", restoreName, " in napespace: ", restore.Cfg.Namespace)
+	} else {
+		restoreName = restore.Cfg.RestoreName
+	}
+
+	kconfig, err := clientcmd.BuildConfigFromFlags("", restore.Cfg.Kubeconfig)
+	if err != nil {
+		logger.Error(err, "Failed to create KubeConfig")
+		return err
+	}
+	logger.Info("Found Kubeconfig", "KUBECONFIG", restore.Cfg.Kubeconfig)
+
+	client, err := kubernetes.NewForConfig(kconfig)
```

Comment:
Nit: You could use the typed client here. It's a bit more comfortable to use than the rest client. See here for an example: https://github.com/k8up-io/k8up/blob/master/restic/kubernetes/config.go#L45

Reply:
we can make a story for that later, at this moment I don't see any advantage to use it